File: backend/app/models/task.py
from dataclasses import dataclass
from datetime import datetime


@dataclass
class Task:
    id: str
    roadmap_week_id: str
    title: str
    description: str | None
    requirements: list[str]
    acceptance_criteria: list[str]
    difficulty: str | None
    estimated_hours: float | None
    resources: list[dict]
    deadline: datetime | None
    status: str

File: frontend/src/types/task.ts
export type TaskDifficulty = "beginner" | "intermediate" | "advanced";
export type TaskStatus = "not_started" | "in_progress" | "submitted" | "completed";

export interface TaskResource {
  label: string;
  url: string;
}

export interface Task {
  id: string;
  roadmap_week_id: string;
  title: string;
  description: string | null;
  requirements: string[];
  acceptance_criteria: string[];
  difficulty: TaskDifficulty | null;
  estimated_hours: number | null;
  resources: TaskResource[];
  deadline: string | null;
  status: TaskStatus;
  created_at: string;
}
